pkgs/by-name/bt/btrfs-list/package.nix

```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  btrfs-progs,
+  perl,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "btrfs-list";
+  version = "2.3";
+
+  src = fetchFromGitHub {
+    owner = "speed47";
+    repo = finalAttrs.pname;
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-cWzDRop0cyrjVIJzuZxTqELgec66GiPAUJY1xIBr3OY=";
+  };
+
+  buildInputs = [ btrfs-progs ];
+
+  patchPhase = ''
+    substituteInPlace btrfs-list --replace-fail /usr/bin/perl ${lib.getExe perl}
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp btrfs-list $out/bin/
```

Comment:
```suggestion
    install -Dm755 btrfs-list -t $out/bin
```

Reply:
What's the advantage of doing it this way? I had it that way initially thinking that `install` was atomic, but it actually isn't AFAICT.

Also why do we need write permissions on the file we put in the store?

Reply:
> What's the advantage of doing it this way? I had it that way initially thinking that `install` was atomic, but it actually isn't AFAICT.

At least shorter?

> Also why do we need write permissions on the file we put in the store?

Not a big deal in both ways, as most packages installed by the build system has write permission.